lxd/auth/authorization_objects.go

```diff
@@ -0,0 +1,208 @@
+// given, or if the correct number of arguments is not passed in. Project objects use the "projectName" argument.
+func ObjectByType(objectType ObjectType, projectName string, args ...string) (string, error) {
+	switch objectType {
+	case ObjectTypeServer:
+		return ObjectServer(), nil
+	case ObjectTypeCertificate:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing certificate fingerprint argument")
+		}
+
+		return ObjectCertificate(args[0]), nil
+	case ObjectTypeStoragePool:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing storage pool name argument")
+		}
+
+		return ObjectStoragePool(args[0]), nil
+	case ObjectTypeProject:
+		return ObjectProject(projectName), nil
+	case ObjectTypeImage:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing image fingerprint argument")
+		}
+
+		return ObjectImage(projectName, args[0]), nil
+	case ObjectTypeImageAlias:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing image alias name argument")
+		}
+
+		return ObjectImageAlias(projectName, args[0]), nil
+	case ObjectTypeInstance:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing instance name argument")
+		}
+
+		return ObjectInstance(projectName, args[0]), nil
+	case ObjectTypeNetwork:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing network name argument")
+		}
+
+		return ObjectNetwork(projectName, args[0]), nil
+	case ObjectTypeNetworkACL:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing network ACL name argument")
+		}
+
+		return ObjectNetworkACL(projectName, args[0]), nil
+	case ObjectTypeNetworkZone:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing network zone name argument")
+		}
+
+		return ObjectNetworkZone(projectName, args[0]), nil
+	case ObjectTypeProfile:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing profile name argument")
+		}
+
+		return ObjectProfile(projectName, args[0]), nil
+	case ObjectTypeStorageBucket:
+		if len(args) != 2 {
+			return "", fmt.Errorf("Missing required arguments for a storage bucket object")
+		}
+
+		return ObjectStorageBucket(projectName, args[0], args[1]), nil
+	case ObjectTypeStoragePoolVolume:
+		if len(args) != 3 {
+			return "", fmt.Errorf("Missing required arguments for a storage pool volume object")
+		}
+
+		return ObjectStoragePoolVolume(projectName, args[0], args[1], args[2]), nil
+	}
+
+	return "", fmt.Errorf("Unknown object type %q", objectType)
+}
+
+// ObjectFromRequest returns an object created from the request by evaluating the given mux vars.
+// Mux vars must be provided in the order that they are found in the endpoint path. If the object
+// requires a project name, this is taken from the project query parameter unless the URL begins
+// with /1.0/projects.
+func ObjectFromRequest(r *http.Request, objectType ObjectType, muxVars ...string) (string, error) {
+	muxValues := make([]string, len(muxVars))
+	for i, muxVar := range muxVars {
+		muxValue, err := url.PathUnescape(mux.Vars(r)[muxVar])
```

Comment:
Can you extract `mux.Vars(r)` once above rather than extracting it for every muxVars item?